runner/src/server/plugins/engine/models/submission/WebhookModel.ts

```diff
@@ -1,90 +1,102 @@
-import { DetailItem } from "../types";
 import { format } from "date-fns";
 import config from "server/config";
+import { FormModel } from "server/plugins/engine/models";
+import { FormSubmissionState } from "server/plugins/engine/types";
+import { FeesModel } from "server/plugins/engine/models/submission/FeesModel";
+import { FormComponent } from "server/plugins/engine/components";
+import { Field } from "server/schemas/types";
+import { PageControllerBase } from "server/plugins/engine/pageControllers";
+import { SelectionControlField } from "server/plugins/engine/components/SelectionControlField";
 import nunjucks from "nunjucks";
+export function WebhookModel(model: FormModel, state: FormSubmissionState) {
+  let englishName = `${config.serviceName} ${model.basePath}`;
 
-function answerFromDetailItem(item) {
-  switch (item.dataType) {
-    case "list":
-      return item.rawValue;
-    case "date":
-      return format(new Date(item.rawValue), "yyyy-MM-dd");
-    case "monthYear":
-      const [month, year] = Object.values(item.rawValue);
-      return format(new Date(`${year}-${month}-1`), "yyyy-MM");
-    default:
-      return item.value;
+  if (model.name) {
+    englishName = model.name.en ?? model.name;
   }
-}
 
-function detailItemToField(item: DetailItem) {
-  return {
-    key: item.name,
-    title: item.title,
-    type: item.dataType,
-    answer: answerFromDetailItem(item),
-  };
-}
+  let questions;
 
-export function WebhookModel(
-  relevantPages,
-  details,
-  model,
-  fees,
-  contextState
-) {
-  const questions = relevantPages?.map((page) => {
-    const isRepeatable = !!page.repeatField;
+  const { relevantPages } = model.getRelevantPages(state);
 
-    const itemsForPage = details.flatMap((detail) =>
-      detail.items.filter((item) => item.path === page.path)
-    );
+  questions = relevantPages.map((page) => pagesToQuestions(page, state));
+  const fees = FeesModel(model, state);
 
-    const detailItems = isRepeatable
-      ? [itemsForPage].map((item) => ({ ...item, isRepeatable }))
-      : itemsForPage;
+  return {
+    metadata: model.def.metadata,
+    name: englishName,
+    questions: questions,
+    ...(!!fees && { fees }),
+  };
+}
 
-    let index = 0;
-    const fields = detailItems.flatMap((item, i) => {
-      item.isRepeatable ? (index = i) : 0;
-      const fields = [detailItemToField(item)];
+function createToFieldsMap(state: FormSubmissionState) {
+  return function (component: FormComponent | SelectionControlField): Field {
```

Comment:
Should this be `FormComponent | SelectionControlField | ComponentCollection`? Just thinking that below you're looking for a children collection, so wasn't sure whether that was related to `ComponentCollection` or whether it's somewhere in the `SelectionControlField` that I couldn't see